game-core/src/test/java/games/strategy/triplea/ui/panels/map/MapRouteDrawerTest.java

```diff
@@ -101,4 +107,17 @@ void testCorrectParameterHandling() {
     verify(dummyRouteDescription, times(2)).getRoute();
     verify(dummyRouteDescription.getRoute(), atLeastOnce()).getAllTerritories();
   }
+
+  /** Regression test for https://github.com/triplea-game/triplea/issues/7112 */
+  @Test
+  void verifySequenceIsTrulyMonotonic() {
+    final MapRouteDrawer routeDrawer = new MapRouteDrawer(mock(MapPanel.class), dummyMapData);
+    final double[] index =
+        routeDrawer.newParameterizedIndex(
+            new Point2D[] {new Double(0, 0), new Double(0, 0), new Double(0, 0)});
```

Comment:
I think we might want to make this test parameterized and include more inputs. In this case we can assert that the output array is always equal in length to the input array and second the index checks can be converted to a simple for loop that verifies each element is less than the next.

Some interesting test cases to consider:
- zero length array
- singleton array (with zero)
- singleton array with negative
- singleton array with positive
- duplicate values other than zero
- out of order values
- mixture of positive and then negative values and then positive values (IE: -1, 1, -1, 1, 0, 1, -1, 0, -1)